com.playeveryware.eos/Runtime/Core/Config/ProductConfig.cs

```diff
@@ -114,6 +141,86 @@ protected override bool NeedsMigration()
         
         protected ProductConfig() : base("eos_product_config.json") { }
 
+        protected override void OnReadCompleted()
+        {
+            // This tracks whether there is a single deployment defined.
+            _deploymentDefined = Environments.IsDeploymentDefined;
+        }
+
+        // This compile conditional is here because the OnWriteCompleted method
+        // is only defined in the editor - so it can only be overriden if in the
+        // editor.
+#if UNITY_EDITOR
+
+        protected override void BeforeWrite()
+        {
+            // If there is one deployment, and one sandbox, then make sure they
+            // are linked to each other.
+            // But only do this if they have been newly added
+            if (!_deploymentDefined && Environments.Deployments.Count ==1 && Environments.Sandboxes.Count == 1)
+            {
+                Environments.Deployments[0].Value.SandboxId = Environments.Sandboxes[0].Value;
```

Comment:
Should this only assign if `Environments.Sandboxes[0]` is not Empty?

Reply:
Yep - good catch. I've updated the implementation. Marking this conversation as resolved accordingly.